src/components/CodingAgentPanel.tsx

```
@@ -0,0 +1,339 @@
+  const [status, setStatus] = useState<AgentStatus | null>(null);
+  const [runs, setRuns] = useState<Run[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [busy, setBusy] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [expanded, setExpanded] = useState<string | null>(null);
+
+  // `load` must not re-run because a translation function was re-created: a
+  // refetch on every render would overwrite a freshly toggled switch with the
+  // stale status it read a moment earlier. Read `t` through a ref instead.
+  const tRef = useRef(t);
+  tRef.current = t;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/components/CodingAgentPanel.tsx:125:3
  123 |   // stale status it read a moment earlier. Read `t` through a ref instead.
  124 |   const tRef = useRef(t);
> 125 |   tRef.current = t;
      |   ^^^^^^^^^^^^ Cannot update ref during render
  126 |
  127 |   const load = useCallback(async () => {
  128 |     try {

(react-hooks/refs)